### ISSUE TITLE
Service Worker Path Is Incorrect After Yarn Build On Windows Server

### ISSUE THREAD

**Issue by ashscodes** [facebook/docusaurus#3420]
## 🐛 Bug Report

We are running 2.0.0-alpha.63 and have just added your PWA plugin.

We serve a production build of Docusaurus via IIS. Our docusaurus build is pulled from our GitLab server to a build directory on the Windows server where we run the relevant yarn commands (install > build). The contents of the build directory that is output by yarn build is copied to the folder where the IIS site as it's home directory. The issue is that during the docusaurus build process, it is adding the drive letter to the sw.js code in one of the compiled scripts.

`function c(e){var t=JSON.stringify(e),n="I:\\sw.js?params="+encodeURIComponent(t);return console.log("[Docusaurus-PWA][registerSW]: service worker url",{url:n,params:e}),n}`

I see the following error in the browser console because the Service Worker doesn't start.

`Uncaught (in promise) TypeError: Failed to register a ServiceWorker: The URL protocol of the script ('i:\sw.js?params=%7B%22offlineMode%22%3Atrue%2C%22debug%22%3Atrue%7D') is not supported.`

Here is the pertinent parts of my docusaurus config and manifest.json

**docusaurus.config.js**
```javascript
plugins: [
    [
      "@docusaurus/plugin-pwa",
      {
        debug: true,
        offlineModeActivationStrategies: ["always"],
        pwaHead: [
          {
            tagName: "link",
            rel: "icon",
            href: "/img/file_system.png",
          },
          {
            tagName: "link",
            rel: "manifest",
            crossorigin: "use-credentials",
            href: "/manifest.json",
          },
          {
            tagName: "meta",
            name: "theme-color",
            content: "rgb(255,73,73)",
          },
          {
            tagName: "meta",
            name: "apple-mobile-web-app-capable",
            content: "yes",
          },
          {
            tagName: "meta",
            name: "apple-mobile-web-app-status-bar-style",
            content: "#000",
          },
          {
            tagName: "link",
            rel: "apple-touch-icon",
            href: "/img/file_system.png",
          },
          {
            tagName: "link",
            rel: "mask-icon",
            href: "/img/file_system.svg",
            color: "rgb(255,88,88)",
          },
          {
            tagName: "meta",
            name: "msapplication-TileImage",
            content: "/img/file_system.png",
          },
          {
            tagName: "meta",
            name: "msapplication-TileColor",
            content: "#000",
          },
        ],
      },
    ],
  ]
```

**static/manifest.json**
```json
{
  "name": "docs",
  "short_name": "docs",
  "description": "site description",
  "theme_color": "#2196f3",
  "background_color": "#424242",
  "display": "standalone",
  "scope": "/",
  "start_url": "/",
  "icons": [
    {
      "src": "/img/file_system-72x72.png",
      "sizes": "72x72",
      "type": "image/png"
    },
    {
      "src": "/img/file_system-96x96.png",
      "sizes": "96x96",
      "type": "image/png"
    },
    {
      "src": "/img/file_system-128x128.png",
      "sizes": "128x128",
      "type": "image/png"
    },
    {
      "src": "/img/file_system-144x144.png",
      "sizes": "144x144",
      "type": "image/png"
    },
    {
      "src": "/img/file_system-152x152.png",
      "sizes": "152x152",
      "type": "image/png"
    },
    {
      "src": "/img/file_system-192x192.png",
      "sizes": "192x192",
      "type": "image/png"
    },
    {
      "src": "/img/file_system-384x384.png",
      "sizes": "384x384",
      "type": "image/png"
    },
    {
      "src": "/img/file_system-512x512.png",
      "sizes": "512x512",
      "type": "image/png"
    }
  ]
}
```

### Have you read the [Contributing Guidelines on issues](https://github.com/facebook/docusaurus/blob/master/CONTRIBUTING.md#reporting-new-issues)?

Yes

## To Reproduce

1. Added `"@docusaurus/plugin-pwa": "^2.0.0-alpha.63"` to `package.json`
2. Updated and pulled site files from GitLab to Windows Server directory for build.
3. Open Command Prompt
4. `I:` (Switch drive)
5. `cd I:\DocsBuild`
6. `yarn install`
7. `yarn build`

Check file that contains function above.

## Expected behavior

When we run this locally using `yarn serve` the service worker starts.

If I modify the JS function to read:

`function c(e){var t=JSON.stringify(e),n="/sw.js?params="+encodeURIComponent(t);return console.log("[Docusaurus-PWA][registerSW]: service worker url",{url:n,params:e}),n}`

The service worker is registered and the PWA works proving that the sw.js script is in the right place and the manifest is correct. This isn't something we can do manually every time the site builds.

## Actual Behavior

Chrome console reports that the URL of the script is not supported.

`Uncaught (in promise) TypeError: Failed to register a ServiceWorker: The URL protocol of the script ('i:\sw.js?params=%7B%22offlineMode%22%3Atrue%2C%22debug%22%3Atrue%7D') is not supported.`

<img width="1035" alt="Screenshot 2020-09-07 at 16 18 24" src="https://user-images.githubusercontent.com/70372500/92405029-ae19c180-f12c-11ea-8276-57d826ddc9bb.png">

Something within the build process is adding the Windows drive letter too.

## Your Environment

- Docusaurus version used: 2.0.0-alpha.63
- Environment name and version (e.g. Chrome 78.0.3904.108, Node.js 10.17.0): **Brave:** Version 1.13.82 Chromium: 85.0.4183.83 (Official Build) (64-bit), **Chrome:** Version 84.0.4147.105 (Official Build) (64-bit)
- Operating system and version (desktop or mobile): Windows Server 2019 Datacenter 1809 (Desktop), IIS 10

## Reproducible Demo

The site isn't hosted anywhere publicly and I am unable to do that. To reproduce, you would need Windows 2019 server and follow the steps **To Reproduce**.

**Comment by slorber** [comment]
Thanks for reporting the issue.
Unfortunately I don't have any windows machine to be able to reproduce this easily, so any help would be appreciated

**Comment by ashscodes** [comment]
Thanks for the response. I appreciate it has an additional requirement in reproducing the issue. Do you know which part of the build process created that function at least? I can't go on record as saying I will find a fix, but I can take a look at where the function is created to see if there is anything I can see and try a few things in our environment.

**Comment by ashscodes** [comment]
I think I found the issue, @slorber, after looking at the source for [docusaurus-plugin-pwa](https://github.com/facebook/docusaurus/blob/master/packages/docusaurus-plugin-pwa/src/index.js). It's using `path.resolve` which adds the drive letter on Windows, so it will never be the relative path on the built site. [Node.js Path docs](https://nodejs.org/api/path.html#path_path)

Suggest changing [this line](https://github.com/facebook/docusaurus/blob/master/packages/docusaurus-plugin-pwa/src/index.js#L76) from:

```javascript
PWA_SERVICE_WORKER_URL: path.resolve(
```

to 

```javascript
PWA_SERVICE_WORKER_URL: path.posix.resolve(
```

This will accommodate people who use the PWA and build on Windows operating systems. I have tested this on MacOS too and it doesn't break anything there on a Unix based system.

Would be happy to submit a PR. Not completed one before, but, in my limited expertise, it's a very small fix.

**Comment by ashscodes** [comment]
Thought I would try and submit this. #3431

**Comment by ashscodes** [comment]
Fixed in #3436 and 2.0.0alpha-64
